src/compiler/scala/tools/nsc/backend/jvm/BCodeHelpers.scala

```diff
@@ -235,22 +234,18 @@ abstract class BCodeHelpers extends BCodeIdiomatic {
   /**
    * This is a hack to work around scala/bug#9111. The completer of `methodSym` may report type errors. We
    * cannot change the typer context of the completer at this point and make it silent: the context
-   * captured when creating the completer in the namer. However, we can temporarily replace
-   * global.reporter (it's a var) to store errors.
+   * captured when creating the completer in the namer. However, we can temporarily ask
+   * global.reporter to suppress reportage.
    */
   def completeSilentlyAndCheckErroneous(sym: Symbol): Boolean =
     if (sym.hasCompleteInfo) false
     else {
-      withoutReporting(sym.info)
-      sym.isErroneous
+      reporter.silently {
+        sym.info
+        sym.isErroneous
```

Comment:
@lrytz I haven't thought about this, but the REPL has so many ways to go silent.